application/forms/SourceForm.php

```diff
@@ -66,23 +66,35 @@ protected function assemble(): void
             'type',
             [
                 'required'          => true,
-                'class'             => 'autosubmit',
                 'label'             => $this->translate('Source Type'),
                 'options'           => $types,
                 'disabledOptions'   => ['']
             ]
         );
 
-        if ($this->getValue('type') !== 'icinga2') {
-            $this->addElement(
-                'text',
-                'listener_username',
-                [
-                    'required' => true,
-                    'label' => $this->translate('Listener Username'),
-                ]
-            );
-        }
+        $this->addElement(
+            'text',
+            'listener_username',
+            [
+                'required' => true,
+                'label' => $this->translate('Listener Username'),
+                'validators' => [new CallbackValidator(
+                    function ($value, CallbackValidator $validator) {
+                        // Deleted rows are included to avoid integrity constraint violations
+                        $source = Source::on($this->db)
+                            ->filter(Filter::equal('listener_username', $value))
+                            ->filter(Filter::any(Filter::equal('deleted', 'y'), Filter::equal('deleted', 'n')))
```

Comment:
Huh, I wasn't aware of that (https://github.com/Icinga/icinga-notifications/pull/338#issuecomment-3520965617) and I think we should set `listener_username` to `NULL` upon deletion.